### ISSUE TITLE
Is there a nice way to do 2D embedding?

### ISSUE THREAD

**Issue by hx364** [fchollet/keras#1693]
Hi Guys,

I am trying to find a way to do a 2D embedding layer.

I am working on sentiment classification on document level. Different from sentiment classification on sentence level, each doc may contain multiple sentences. The idea is to use a convolution layer to extract a sentence representation for each sentence, then feed the sentence representation into a LSTM/GRU for classification. It's very similar to the fig1 in this paper: http://arxiv.org/pdf/1411.4389v3.pdf, except that the prediction is only made at the last time step.

The embedding layer in keras takes a input of  2D tensor, and ouput  a 3D tensor. However, after padding, my X is a 3D tensor of shape (nb_samples, maxlen_doc, maxlen_sentence), and want to transform it via a Embedding layer to shape (nb_sample, maxlen_doc, maxlen_sentence, embedding_dims).

I was wondering are there any nice ways to do so? Or I need to create a 2Dembedding class? Thanks!

**Comment by wxs** [comment]
Just reshape your `X` to 2d, use a normal embedding, and then reshape its output via http://keras.io/layers/core/#reshape

**Comment by hx364** [comment]
Thanks! very nice way to do so.
